src/gsi/read_gnssrspd.f90

```diff
@@ -0,0 +1,519 @@
+     character(40) :: obs_region  
+     character( 8) :: subset
+     character( 8) :: c_prvstg,c_sprvstg
+     character( 8) :: c_station_id
+     character( 6) :: bulstr1,bulstr2  
+     character( 6) :: obsbul(2,1)  
+     character(10) date
+!    Integer variables
+     integer(i_kind), parameter :: mxib  = 31
+     integer(i_kind), parameter :: ietabl= 19 
+
+     integer(i_kind) :: i,k,kl,k1,k2,j 
+     integer(i_kind) :: ihh,idd,idate,iret,im,iy,levs
+     integer(i_kind) :: lunin 
+     integer(i_kind) :: ireadmg,ireadsb
+     integer(i_kind) :: ilat,ilon 
+     integer(i_kind) :: nlv
+     integer(i_kind) :: nreal,nchanl
+     integer(i_kind) :: idomsfc,isflg
+     integer(i_kind) :: ithin,iout 
+     integer(i_kind) :: nc,ncsave
+     integer(i_kind) :: ntmatch,ntb
+     integer(i_kind) :: nmsg   
+     integer(i_kind) :: maxobs 
+     integer(i_kind) :: itype,itypey,iecol
+     integer(i_kind) :: ierr_ps,ierr_q,ierr_t,ierr_uv,ncount_ps,ncount_q,ncount_t,ncount_uv 
+     integer(i_kind) :: qcm,lim_qm
+     integer(i_kind) :: p_qm,g_qm,t_qm,q_qm,uv_qm,wspd_qm,ps_qm
+     integer(i_kind) :: ntest,nvtest
+!    integer(i_kind) :: m,itypex,lcount,iflag
+     integer(i_kind) :: nlevp   ! vertical level for thinning
+     integer(i_kind) :: pflag   
+     integer(i_kind) :: ntmp,iiout,igood
+     integer(i_kind) :: kk,klon1,klat1,klonp1,klatp1
+     integer(i_kind) :: iuse
+     integer(i_kind) :: nmind
+     integer(i_kind) :: nib 
+ 
+     integer(i_kind) :: ibit(mxib)
+     integer(i_kind) :: idate5(5)
+     integer(i_kind) :: minobs,minan
+
+     integer(i_kind), allocatable,dimension(:) :: isort
+
+!    Real variables
+     real(r_kind), parameter :: r0_001  =  0.001_r_kind
+     real(r_kind), parameter :: r1_2    =    1.2_r_kind
+     real(r_kind), parameter :: r3_0    =    3.0_r_kind
+     real(r_kind), parameter :: r0_7    =    0.7_r_kind
+     real(r_kind), parameter :: r6      =    6.0_r_kind
+     real(r_kind), parameter :: r50     =   50.0_r_kind
+     real(r_kind), parameter :: r1200   = 1200.0_r_kind
+     real(r_kind), parameter :: emerr   =    0.2_r_kind ! RH
+     real(r_kind), parameter :: missing = 1.0e+11_r_kind
+     real(r_kind), parameter :: r180    = 180.0_r_kind
+     real(r_kind), parameter :: r360    = 360.0_r_kind
+
+     real(r_kind) :: toff,t4dv
+     real(r_kind) :: rmesh
+     real(r_kind) :: usage
+     real(r_kind) :: woe,toe,qoe,psoe,obserr,var_jb
+     real(r_kind) :: dlat,dlon,dlat_earth,dlon_earth
+     real(r_kind) :: dlat_earth_deg,dlon_earth_deg
+     real(r_kind) :: cdist,disterr,disterrmax,rlon00,rlat00
+     real(r_kind) :: vdisterrmax,u00,v00,u0,v0
+     real(r_kind) :: dx,dy,dx1,dy1,w00,w10,w01,w11
+     real(r_kind) :: wdir,wspd
+     real(r_kind) :: tob,uob,vob,qob,spdob,rrob
+     real(r_kind) :: rhob,tdob,dlnpsob,log
+     real(r_kind) :: pob_mb,pob_cb,pob_pa,gob
+     real(r_kind) :: psob_mb,psob_cb,psob_pa
+     real(r_kind) :: qmaxerr 
+     real(r_kind) :: dlnpsob,dlnpob,ppb
```

Comment:
`dlnpsob` was declared on line 157.  Either remove the `dlnpsob` declaration on this line or line 157.  The code does not compile with two declarations of the same variable.  

Notice that `ppb` is not used in this routine.  Please check the routines you added or modified and remove unused variables.

Reply:
Removed!